subsys/bluetooth/host/gatt.c

```diff
@@ -4587,13 +4592,41 @@ int bt_gatt_resubscribe(uint8_t id, const bt_addr_le_t *peer,
 	return 0;
 }
 
+static bool sub_has_subscription(struct gatt_sub *sub,
+	const struct bt_gatt_subscribe_params *params)
+{
+	struct bt_gatt_subscribe_params *other_params;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&sub->list, other_params, node) {
+		if (params == other_params) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool sub_has_other_subscriptions_to_same_attribute(struct gatt_sub *sub,
+	const struct bt_gatt_subscribe_params *params)
+{
+	struct bt_gatt_subscribe_params *other_params;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&sub->list, other_params, node) {
+		if (params != other_params) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
 int bt_gatt_unsubscribe(struct bt_conn *conn,
 			struct bt_gatt_subscribe_params *params)
 {
 	struct gatt_sub *sub;
 	struct bt_gatt_subscribe_params *tmp, *next;
-	bool has_subscription = false, found = false;
 	sys_snode_t *prev = NULL;
+	bool wait_for_ccc_write_response = false;
```

Comment:
How about calling this `other_sub' and not pre-initializing it, rather pass the return value from `has_other_sub()` to it?